File: evidence/generated-systems/angular/healthcare-on-angular/src/app/models/Clinician.js

// Define collection and schema for Clinician
export interface Clinician {
    firstName:
	type : string
    lastName:
	type : string
    licenseNumber:
	type : string
    CareTeams:
 	type : [{ type: Schema.Types.ObjectId, ref: 'CareTeam' }]
    Appointments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Appointment' }]
    Encounters:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Encounter' }]
    Procedures:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Procedure' }]
    ImagingReports:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ImagingReport' }]
    ClinicianType:
 	type : String
    Specialty:
 	type : String
#
    collection: 'clinicians'
}

File: evidence/generated-systems/golang/healthcare-on-golang/healthcare-on-golang/internal/model/Clinician.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Clinician Declaration
//==============================================================
type Clinician struct {
    gorm.Model
     FirstName                                    string
    LastName                                    string
    LicenseNumber                                    string
     CareTeams           []CareTeam `gorm:"foreignKey:CareTeamsFromClinicianId"`
     Appointments           []Appointment `gorm:"foreignKey:AppointmentsFromClinicianId"`
     Encounters           []Encounter `gorm:"foreignKey:EncountersFromClinicianId"`
     Procedures           []Procedure `gorm:"foreignKey:ProceduresFromClinicianId"`
     ImagingReports           []ImagingReport `gorm:"foreignKey:ImagingReportsFromClinicianId"`
    ClinicianType                      ClinicianType
    Specialty                      ClinicianSpecialty

// parent associations as their child

}

